src/truncate.js

```diff
@@ -1,41 +1,49 @@
-function registerTruncateMagicMethod() {
-    Alpine.addMagicProperty('truncate', function ($el) {
-        return (...parameters) => {
-            if (typeof parameters[0] !== 'string') return parameters[0]
-            let ellipsis = '…'
-
-            // If the second parameter isn't truthy, return the full string
-            if (!parameters[1]) return parameters[0]
-
-            // if only a number or string is passed in, keep it simple
-            if (typeof parameters[1] !== 'object') {
-                if (typeof parameters[2] !== 'undefined') {
-                    ellipsis = parameters[2]
+import { checkForAlpine } from './utils'
+
+const AlpineTruncateMagicMethod = {
+    start() {
+        checkForAlpine()
+
+        Alpine.addMagicProperty('truncate', function () {
+            return (...parameters) => {
```

Comment:
maybe beyond the scope of this PR, but are we really using a variable number of parameters?

Pretty sure it's always `(string, options)`? would make the code a bit more readable as well

Reply:
Just saw this. It was originally designed with an optional 3rd param to override the "..." that's appended. But we could make it `(string, options)` though and keep that format throughout all the helpers.

Reply:
https://github.com/alpine-collective/alpine-magic-helpers/issues/16